legacyPost/src/main/java/com/hamdan/forzenbook/legacy/post/LegacyPostActivity.kt

```diff
@@ -1,33 +1,196 @@
-        binding.postLayoutToolBar.setOnMenuItemClickListener {
-            when (it.itemId) {
-                R.id.postSendIcon -> {
-                    Log.v("Hamdan", "CLICKED IT")
-                    true
+        binding.apply {
+            val setPill: (Boolean) -> Unit = {
+                if (it) {
+                    pillLeftFrame.isEnabled = true
+                    textToggle.isChecked = true
+                    pillRightFrame.isEnabled = false
+                    imageToggle.isChecked = false
+                } else {
+                    inputPostText.setText("")
+                    pillLeftFrame.isEnabled = false
+                    textToggle.isChecked = false
+                    pillRightFrame.isEnabled = true
+                    imageToggle.isChecked = true
+                }
+                postViewModel.toggleClicked()
+            }
+            val showImageContent: (Boolean) -> Unit = { showBottomButton ->
+                postLoadIndicator.isVisible = false
+                if (showBottomButton) {
+                    postChooseImage.isVisible = true
+                    scrollPost.isVisible = false
+                    postChosenImage.isVisible = false
+                    postChangeImage.isVisible = false
+                } else {
+                    postChooseImage.isVisible = false
+                    scrollPost.isVisible = false
+                    postChosenImage.isVisible = true
+                    postChangeImage.isVisible = true
+                }
+            }
+            val showTextContent = {
+                postLoadIndicator.isVisible = false
+                postChooseImage.isVisible = false
+                scrollPost.isVisible = true
+                postChosenImage.isVisible = false
+                postChangeImage.isVisible = false
+            }
+            val loadingContent = {
+                postLoadIndicator.isVisible = true
+                postChooseImage.isVisible = false
+                scrollPost.isVisible = false
+                postChosenImage.isVisible = false
+                postChangeImage.isVisible = false
+            }
+            val errorContent = {
+                postLoadIndicator.isVisible = false
+                postChooseImage.isVisible = false
+                scrollPost.isVisible = false
+                postChosenImage.isVisible = false
+                postChangeImage.isVisible = false
+            }
+            postLayoutToolBar.inflateMenu(R.menu.post_menu)
+            postLayoutToolBar.setOnMenuItemClickListener { item ->
+                when (item.itemId) {
+                    R.id.postSendIcon -> {
+                        postViewModel.sendPostClicked(this@LegacyPostActivity)
+                        true
+                    }
+
+                    else -> {
+                        false
+                    }
+                }
+            }
+            pillLeftFrame.isEnabled = true
+            textToggle.isChecked = true
```

Comment:
You are holding too much state management code in this Activity. The VM needs to handle the state. Will this really all be restored on rotation correctly? Remember that the View needs to be dumb and a subscription to VM state needs to be all that updates the View fields

Reply:
You're 100 percent right, I forgot about rotation and I was feeling like something was missing from it when I was thinking it was done. it was a simple fix though to read from the view model for the state the view should be in.